rgw/v2/tests/s3cmd/test_large_object_gc.py

```diff
@@ -0,0 +1,173 @@
+            raise TestExecError("RGW service restart failed")
+        else:
+            log.info("RGW service restarted")
+        log.info("download large object again to make gc list with shadow entries")
+
+        s3cmd_reusable.download_file(
+            bucket_name, uploaded_file, local_file_name="download1.img"
+        )
+        time.sleep(5)
+        downloaded_file1_md5 = utils.get_md5("download1.img")
+        assert uploaded_file_md5 == downloaded_file1_md5
+        gc_list_output = json.loads(
+            utils.exec_shell_cmd("radosgw-admin gc list --include-all")
+        )
+
+        log.info(gc_list_output)
+
+        if gc_list_output:
+            log.info("Shadow obj found after setting rgw_gc_obj_min_wait to 5 sec")
+            utils.exec_shell_cmd("radosgw-admin gc process --include-all")
+            log.info("Object download should not error out in 404 NoSuchKey error")
+            s3cmd_reusable.download_file(
+                bucket_name, uploaded_file, local_file_name="download2.img"
+            )
+            downloaded_file2_md5 = utils.get_md5("download2.img")
+            assert uploaded_file_md5 == downloaded_file2_md5
+
+    # Delete file from bucket
+    s3cmd_reusable.delete_file(bucket_name, uploaded_file)
+    log.info(f"Deleted file {uploaded_file} from bucket {bucket_name}")
+
+    # Delete bucket
+    s3cmd_reusable.delete_bucket(bucket_name)
+    log.info(f"Bucket {bucket_name} deleted")
+
+    # check for any crashes during the execution
+    crash_info = reusable.check_for_crash()
+    if crash_info:
+        raise TestExecError("ceph daemon crash found!")
+
+
+if __name__ == "__main__":
+
+    test_info = AddTestInfo("test swift user key gen")
+
+    try:
+        project_dir = os.path.abspath(os.path.join(__file__, "../../.."))
+        test_data_dir = "test_data"
+        TEST_DATA_PATH = os.path.join(project_dir, test_data_dir)
+        log.info(f"TEST_DATA_PATH: {TEST_DATA_PATH}")
+        if not os.path.exists(TEST_DATA_PATH):
+            log.info("test data dir not exists, creating.. ")
+            os.makedirs(TEST_DATA_PATH)
+        parser = argparse.ArgumentParser(
+            description="RGW S3 Large Object download using s3cmd"
+        )
+        parser.add_argument("-c", dest="config", help="RGW S3 Large Object using s3cmd")
+        parser.add_argument(
+            "-log_level",
+            dest="log_level",
+            help="Set Log Level [DEBUG, INFO, WARNING, ERROR, CRITICAL]",
+            default="info",
+        )
+        args = parser.parse_args()
+        yaml_file = args.config
+        log_f_name = os.path.basename(os.path.splitext(yaml_file)[0])
+        configure_logging(f_name=log_f_name, set_level=args.log_level.upper())
+        config = resource_op.Config(yaml_file)
+        config.read()
+        test_exec(config)
+        test_info.success_status("test passed")
+        sys.exit(0)
+
+    except (RGWBaseException, Exception) as e:
+        log.info(e)
```

Comment:
Let's cleanup the `TEST_DATA_PATH` directory... this would allow us to avoid cleanup in the lower levels. We could add it in the `finally` section.

Reply:
@psathyan , Implemented this PR comment. Please review the same